components/Video/StoryVideo.tsx

```diff
@@ -0,0 +1,125 @@
+import { useEffect, useRef, useState } from 'react';
+import { cnb } from 'cnbuilder';
+import { useInView } from 'framer-motion';
+import { type AnimationType } from '@/components/Animate';
+import { MediaWrapper, type MediaWrapperProps } from '@/components/Media';
+import { MutedVideoLoop, VideoButton } from '@/components/Video';
+import { type PaddingType } from '@/utilities/datasource';
+import * as styles from './StoryVideo.styles';
+
+/**
+ * Inline story Video component that comes with a play/pause button and layout/caption options
+ * similar to the StoryImage component.
+ * To be used for the SbVideo component in Storyblok.
+ */
+
+export type StoryVideoProps = React.HTMLAttributes<HTMLDivElement> & Omit<MediaWrapperProps, 'isFullHeight' | 'isParallax'> & {
+  videoWebm?: string;
+  videoMp4?: string;
+  videoPosterSrc?: string;
+  isFullScreen?: boolean; // Whether the video takes the full width of the device
+  spacingTop?: PaddingType;
+  spacingBottom?: PaddingType;
+  animation?: AnimationType;
+  delay?: number;
+  aspectRatioClass?: string;
+};
+
+export const StoryVideo = ({
```

Comment:
This is a counterpart of the `StoryImage` component - comes with a wrapper that provides aspect ratio, width and spacing options, plus caption and other options.